### PR TITLE
[TEVA-1758] Add route to jobseeker job applications index page

### PR DESCRIPTION
**Pull request by cpjmcquillan**
## Jira ticket URL
https://dfedigital.atlassian.net/browse/TEVA-1758

## Changes in this PR
- Consistent option order in routes (only/controller)
- Use resource for jobseeker job_application new/create routes (prevents clash + makes more sense)
- Add index route for jobseeker job_applications

### CODE REVIEW

**Comment by martin-bangoura** [comment]
Review app deployed to <https://teaching-vacancies-review-pr-2982.london.cloudapps.digital>

File: config/routes.rb [hunk 1]
```diff
@@ -27,18 +27,18 @@
...
         resources :build, only: %i[show update], controller: "job_applications/build" do
           resources :details, only: %i[new create edit update destroy], controller: "job_applications/details"
         end
         get :review
         post :submit
         resource :feedback, only: %i[create], controller: "job_applications/feedbacks"
       end
+
+      scope path: ":job_id" do
+        resource :job_application, only: %i[new create]
```
Comment (josephhull676) [inline config/routes.rb:40]: Why the change to resource (singular) and job_application (singular)?

Comment (cpjmcquillan) [reply config/routes.rb:40]: > Use resource for jobseeker job_application new/create routes (prevents clash + makes more sense)
It prevents a clash between the create action for a job application and the new route. And it makes more sense because there's never going to be more than one job application for a particular jobseeker for a particular job

Comment (david-mears-dfe) [reply config/routes.rb:40]: My guess: We're in the namespace job_ids within the namespace for jobseekers, and there can only be one job application per job per jobseeker.

**Review by david-mears-dfe — APPROVED** [review]

File: config/routes.rb (hunk 1 above)
Comment (cpjmcquillan) [reply config/routes.rb:40]: I should probably do the same for saved jobs as well, but I'll make another micro PR for that
